### PR TITLE
Grab temple adventures

### PR DESCRIPTION
**Pull request by CheeseyPickle**
Grab Hidden Temple adventures if it's unlocked and you're also ascending.

Probably not the most optimal place to put it because it also extends up to 10 effects, but I have no idea how to make it optimal, and right now the adventures are more important.

### CODE REVIEW

**Comment by horrible-little-slime** [comment]
Consider adding a price check for stone wool--equilibrium price is going to be slightly higher than 3 * valueOfAdventure

File: src/fights.ts [hunk 1]
```diff
@@ -385,6 +385,12 @@ function embezzlerSetup() {
     }
   });
   if (have($item`License to Chill`) && !get("_licenseToChillUsed")) use($item`License to Chill`);
+  if (globalOptions.ascending && get("questM16Temple") === "finished" && get("lastTempleAdventures") < get("knownAscensions")) {
```
Comment (horrible-little-slime) [inline src/fights.ts:388]: consider using questStep in lieu of === "finished", and myAscensions() in lieu of get("knownAscensions")

**Review by horrible-little-slime — APPROVED** [review]
Very good, although I'm not entirely sure we want to run from baa'baa'bur'ann in the unlikely event we hit her. I think that's up in the air, though, and this is strictly speaking better than what we're doing right now, so I'm sold.

File: src/fights.ts [hunk 2]
```diff
@@ -379,6 +380,24 @@ function embezzlerSetup() {
...
+    acquire(1, $item`stone wool`, 3 * get("valueOfAdventure") + 100, false) > 0
+  ) {
+    ensureEffect($effect`Stone-Faced`);
+    setChoice(582, 1);
+    setChoice(579, 3);
+    const runSource = findRun();
+    if (runSource) {
+      if (runSource.prepare) runSource.prepare();
+      freeFightOutfit([...(runSource.requirement ? [runSource.requirement] : [])]);
+      while (get("lastTempleAdventures") < myAscensions()) {
```
Comment (Rinn) [inline src/fights.ts:396]: this isn't going to work, you need to get a new runSource every iteration, prepare it, and equip the requirements

**Review by Rinn — CHANGES_REQUESTED** [review]

File: src/fights.ts (hunk 2 above)
Comment (horrible-little-slime) [reply src/fights.ts:396]: as an addon to this, I think this is one of the situations like with startDigitize() where you may want to add a backup LTB if you can't find a run

**Pull request merged**